pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -151,16 +164,24 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 		// a new object is going to be created
 		if deploy.ObjectMeta.CreationTimestamp.IsZero() {
 			deploy.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: deploymentLabels,
+				MatchLabels: selectorLabels,
 			}
 		}
 
-		// update
-		deploy.ObjectMeta.Labels = deploymentLabels
+		// update the deployment labels
+		objectLabels := make(map[string]string, 0)
+		for key, value := range selectorLabels {
+			objectLabels[key] = value
+		}
+		topology, _ := osd.ExtractRookTopologyFromLabels(node.GetLabels())
+		for key, value := range topology {
+			objectLabels[key] = value
+		}
+		deploy.ObjectMeta.Labels = objectLabels
```

Comment:
Why does the the canary deployment need the labels from the node? The canary has a node selector to the node, so someone can look at the labels of the node anyway.

Reply:
This is so we can delete canaries by failure domain via labelSelector for cleaning up stale canaries.

Identifying when to delete stale canaries turned out to be a little tricky. Currently, the drain logic deletes the canary for the entire failure domain that is being okayed for draining, and the nodedrain controller brings it back if the node becomes scheduled with OSDs again.